staging/src/k8s.io/apiserver/pkg/storage/cacher/lister_watcher.go

```diff
@@ -59,7 +63,11 @@ func (lw *listerWatcher) List(options metav1.ListOptions) (runtime.Object, error
 		Predicate:            pred,
 		Recursive:            true,
 	}
-	if err := lw.storage.GetList(context.TODO(), lw.resourcePrefix, storageOpts, list); err != nil {
+	ctx := context.Background()
+	if lw.contextMetadata != nil {
```

Comment:
Does it matter if lw.contextMetadata is `nil` or empty? (can we skip creating NewOutgoingContext in both cases?)

Reply:
Wanted to maintain the exact old behavior if feature flag is disabled. Haven't confirmed if there is a difference.